### PR TITLE
graph: fix multi-node channel count

### PR DESCRIPTION
**Pull request by nusislam**
## Details
This PR fixes the logic to restrict channel count to 64 for multi-node.

**Work item:** _"Internal", or link to GitHub issue (if applicable)._

**What were the changes?**  
_One sentence describing the work done._

**Why were the changes made?**  
_Explain the motivation behind the work. Provide any publicly-available historical context._

**How was the outcome achieved?**  
_Technical details behind the work. Explain any publicly-available hardware peculiarities._

**Additional Documentation:**  
_What else should the reviewer know?_

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

File: src/graph/connect.cc [hunk 1]
```diff
@@ -672,7 +672,8 @@ ncclResult_t ncclTopoPostset(struct ncclComm* comm, int* firstRanks, int* treePa
   if (comm->nNodes == 1) {
```
Comment (wenkaidu) [inline src/graph/connect.cc]: can you remove minNchannels = 64?

Comment (wenkaidu) [reply src/graph/connect.cc]: or you can minNchannels = ncclMinNchannels(), then simplify if conditions

**Review by wenkaidu — APPROVED** [review]

File: src/graph/connect.cc (hunk 1 above)
Comment (nusislam) [reply src/graph/connect.cc]: fixed

**Comment by nusislam** [comment]
@wenkaidu - Added a new patch. Please review.

File: src/graph/paths.cc [hunk 2]
```diff
@@ -796,9 +796,11 @@ ncclResult_t ncclTopoTrimSystem(struct ncclTopoSystem* system, struct ncclComm*
     INFO(NCCL_GRAPH, "GDR is available on all GPUs");
   }
 
+  comm->hasIntraNet = false;
   if (rcclParamEnableIntranet() == 1) {
     remove = 0;
     system->type |= RCCL_TOPO_FORCE_INTRA;
+    comm->hasIntraNet = true;
```
Comment (wenkaidu) [inline src/graph/paths.cc]: The env var enabled intra-net maybe set without actually enables it. For example, no NIC is available.
https://github.com/ROCm/rccl/blob/develop/src/include/graph.h#L98
nIntraChannels reflects if intra-net is actually enabled

**Comment by wenkaidu** [comment]
> @wenkaidu - Added a new patch. Please review.

please use nIntraChannels

File: src/graph/connect.cc [hunk 3]
```diff
@@ -627,6 +627,10 @@ ncclResult_t ncclTopoPostset(struct ncclComm* comm, int* firstRanks, int* treePa
   // Only use full MAXCHANNELS for gfx94x
   int maxChannels = IsArchMatch(comm->topo->nodes[GPU].nodes[0].gpu.gcn, "gfx94") ? MAXCHANNELS : 2*CHANNEL_LIMIT;
 
+  if (graphs[NCCL_ALGO_RING]->nIntraChannels > 0 || comm->nNodes > 1) {
```
Comment (wenkaidu) [inline src/graph/connect.cc:630]: why nNodes>1? single node is not applicable here?

Comment (nusislam) [reply src/graph/connect.cc:630]: I set `maxChannel=64` for `nNodes>1` or `nIntraChannels >1` (can be true for single node also).

Comment (wenkaidu) [reply src/graph/connect.cc:630]: got it

**Pull request merged**